nixos/tests/resolv.nix

```diff
@@ -0,0 +1,77 @@
+# Test whether DNS resolving returns multiple records and all address families.
+import ./make-test-python.nix ({ pkgs, ... } : {
+  name = "resolv";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ ckauhaus ];
+  };
+
+  nodes.resolv = { ... }: {
+    networking.extraHosts = ''
+      # IPv4 only
+      192.0.2.1 host-ipv4.example.net
+      192.0.2.2 host-ipv4.example.net
+      # IP6 only
+      2001:db8::2:1 host-ipv6.example.net
+      2001:db8::2:2 host-ipv6.example.net
+      # dual stack
+      192.0.2.1 host-dual.example.net
+      192.0.2.2 host-dual.example.net
+      2001:db8::2:1 host-dual.example.net
+      2001:db8::2:2 host-dual.example.net
+    '';
+  };
+
+  testScript = let
+    getaddrinfo_py = pkgs.writeScript "getaddrinfo.py" ''
+      import socket
+      import sys
+
+      result = set()
+      for gai in socket.getaddrinfo(sys.argv[1], 0):
+          result.add(gai[4][0])
```

Comment:
How about this generator approach:

```python
result = {gai[4][0] for gai in socket.getaddrinfo(sys.argv[1], 0)}
```

Reply:
Rewrote test script based on your first suggestion. Please comment. If this version is fine, I'll squash the PR.